lib/src/openApiToZod.ts

```diff
@@ -107,9 +107,22 @@ export function getZodSchema({ schema, ctx, meta: inheritedMeta, options }: Conv
             return code.assign(type.toString());
         }
 
-        const types = schema.anyOf.map((prop) => getZodSchema({ schema: prop, ctx, meta, options })).join(", ");
-        const oneOf = `z.union([${types}])`;
-        return code.assign(`z.union([${oneOf}, z.array(${oneOf})])`);
+        const types = schema.anyOf
+            .map((prop) => getZodSchema({ schema: prop, ctx, meta, options }))
+            .map((type) => {
+                let isPrimitive = false;
+
+                if ("type" in type.schema && !Array.isArray(type.schema.type)) {
```

Comment:
also, what does it mean for the schema's type to be an array? i couldn't tell if this is TS getting the types wrong or not, so i filtered that out. but does it need to be specifically handled somehow?

Reply:
it can happen since OpenAPI 3.1

it was introduced [in this PR](https://github.com/astahmer/openapi-zod-client/commit/519de1409be4dbe6efb2c25db58669b3c48f38b7) because of [this issue](https://github.com/astahmer/openapi-zod-client/issues/60)

Reply:
from what we wrote in this issue:

> it basically acts as a or operator, like the SchemaObject["oneOf"]

Reply:
yes and yes ! it was handled here by indeed just looping on the `type` and then calling `getZodSchema` using a that isolated type if it's a complex type or just using an union otherwise

https://github.com/astahmer/openapi-zod-client/commit/519de1409be4dbe6efb2c25db58669b3c48f38b7#diff-aff08f665d4e69fe50761cef05035287658147285c711089446e5cbbbadacb54R66-R77